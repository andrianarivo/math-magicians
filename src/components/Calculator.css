.calculator {
  display: flex;
  flex-direction: column;
  height: 100%;
}

.row {
  display: flex;
  flex: 1;
  flex-direction: row;
  flex-wrap: wrap;
}
