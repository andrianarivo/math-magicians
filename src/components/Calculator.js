import { useState } from 'react';
import styles from '../styles/Calculator.module.css';
import Button from './Button';
import calculate from '../logic/calculate';

function Calculator() {
  const [calculatedObj, setCalculatedObj] = useState({
    next: null,
    total: null,
    operator: null,
  });
  const handleClick = (buttonName) => {
    const newObj = calculate(calculatedObj, buttonName);
    setCalculatedObj(newObj);
  };

  const resultText = () => {
    const { next, operation, total } = calculatedObj;
    let resultString = '0';
    if (next || operation) {
      resultString = next ?? operation;
    }
    if (total && !next && !operation) {
      resultString = total;
    }
    return resultString;
  };

  return (
    <div className={styles.calculator}>
      <Button
        text={`${resultText()}`}
        textAlign="right"
        color="#858693"
        textColor="#fff"
        cursor="auto"
        testId="result-text-button"
        onClick={() => {}}
      />
      <div className={styles.row}>
        <Button text="AC" onClick={handleClick} />
        <Button text="+/-" onClick={handleClick} />
        <Button text="%" onClick={handleClick} />
        <Button text="รท" color="#f5913e" onClick={handleClick} />
      </div>
      <div className={styles.row}>
        <Button text="7" onClick={handleClick} />
        <Button text="8" onClick={handleClick} />
        <Button text="9" onClick={handleClick} />
        <Button text="x" color="#f5913e" onClick={handleClick} />
      </div>
      <div className={styles.row}>
        <Button text="4" onClick={handleClick} />
        <Button text="5" onClick={handleClick} />
        <Button text="6" onClick={handleClick} />
        <Button text="-" color="#f5913e" onClick={handleClick} />
      </div>
      <div className={styles.row}>
        <Button text="1" onClick={handleClick} />
        <Button text="2" onClick={handleClick} />
        <Button text="3" onClick={handleClick} />
        <Button text="+" color="#f5913e" onClick={handleClick} />
      </div>
      <div className={styles.row}>
        <Button text="0" flex={2} onClick={handleClick} />
        <Button text="." onClick={handleClick} />
        <Button text="=" color="#f5913e" onClick={handleClick} />
      </div>
    </div>
  );
}

Calculator.defaultProps = {};

Calculator.propTypes = {};

export default Calculator;
