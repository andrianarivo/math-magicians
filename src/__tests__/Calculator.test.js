import renderer from 'react-test-renderer';
import { fireEvent, render, screen } from '@testing-library/react';
import Calculator from '../components/Calculator';

describe('testing Calculator component', () => {
  test('render', () => {
    const tree = renderer.create(<Calculator />);
    expect(tree).toMatchSnapshot();
  });

  test('user interaction', () => {
    render(<Calculator />);

    const resultButton = screen.getByTestId('result-text-button');
    fireEvent.click(screen.getByText('1'));

    expect(resultButton.innerHTML).toBe('<p>1</p>');
  });
});
