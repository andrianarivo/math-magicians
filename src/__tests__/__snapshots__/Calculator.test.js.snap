// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing Calculator component render 1`] = `
<div
  className="calculator"
>
  <button
    className="button"
    data-testid="result-text-button"
    onClick={[Function]}
    onKeyDown={[Function]}
    style={
      Object {
        "backgroundColor": "#858693",
        "color": "#fff",
        "cursor": "auto",
        "flex": 1,
        "justifyContent": "flex-end",
        "paddingRight": "2em",
      }
    }
    type="button"
  >
    <p>
      0
    </p>
  </button>
  <div
    className="row"
  >
    <button
      className="button"
      data-testid="button"
      onClick={[Function]}
      onKeyDown={[Function]}
      style={
        Object {
          "backgroundColor": "#e0e0e0",
          "color": "#000",
          "cursor": "pointer",
          "flex": 1,
          "justifyContent": "center",
          "paddingRight": "0",
        }
      }
      type="button"
    >
      <p>
        AC
      </p>
    </button>
    <button
      className="button"
      data-testid="button"
      onClick={[Function]}
      onKeyDown={[Function]}
      style={
        Object {
          "backgroundColor": "#e0e0e0",
          "color": "#000",
          "cursor": "pointer",
          "flex": 1,
          "justifyContent": "center",
          "paddingRight": "0",
        }
      }
      type="button"
    >
      <p>
        +/-
      </p>
    </button>
    <button
      className="button"
      data-testid="button"
      onClick={[Function]}
      onKeyDown={[Function]}
      style={
        Object {
          "backgroundColor": "#e0e0e0",
          "color": "#000",
          "cursor": "pointer",
          "flex": 1,
          "justifyContent": "center",
          "paddingRight": "0",
        }
      }
      type="button"
    >
      <p>
        %
      </p>
    </button>
    <button
      className="button"
      data-testid="button"
      onClick={[Function]}
      onKeyDown={[Function]}
      style={
        Object {
          "backgroundColor": "#f5913e",
          "color": "#000",
          "cursor": "pointer",
          "flex": 1,
          "justifyContent": "center",
          "paddingRight": "0",
        }
      }
      type="button"
    >
      <p>
        รท
      </p>
    </button>
  </div>
  <div
    className="row"
  >
    <button
      className="button"
      data-testid="button"
      onClick={[Function]}
      onKeyDown={[Function]}
      style={
        Object {
          "backgroundColor": "#e0e0e0",
          "color": "#000",
          "cursor": "pointer",
          "flex": 1,
          "justifyContent": "center",
          "paddingRight": "0",
        }
      }
      type="button"
    >
      <p>
        7
      </p>
    </button>
    <button
      className="button"
      data-testid="button"
      onClick={[Function]}
      onKeyDown={[Function]}
      style={
        Object {
          "backgroundColor": "#e0e0e0",
          "color": "#000",
          "cursor": "pointer",
          "flex": 1,
          "justifyContent": "center",
          "paddingRight": "0",
        }
      }
      type="button"
    >
      <p>
        8
      </p>
    </button>
    <button
      className="button"
      data-testid="button"
      onClick={[Function]}
      onKeyDown={[Function]}
      style={
        Object {
          "backgroundColor": "#e0e0e0",
          "color": "#000",
          "cursor": "pointer",
          "flex": 1,
          "justifyContent": "center",
          "paddingRight": "0",
        }
      }
      type="button"
    >
      <p>
        9
      </p>
    </button>
    <button
      className="button"
      data-testid="button"
      onClick={[Function]}
      onKeyDown={[Function]}
      style={
        Object {
          "backgroundColor": "#f5913e",
          "color": "#000",
          "cursor": "pointer",
          "flex": 1,
          "justifyContent": "center",
          "paddingRight": "0",
        }
      }
      type="button"
    >
      <p>
        x
      </p>
    </button>
  </div>
  <div
    className="row"
  >
    <button
      className="button"
      data-testid="button"
      onClick={[Function]}
      onKeyDown={[Function]}
      style={
        Object {
          "backgroundColor": "#e0e0e0",
          "color": "#000",
          "cursor": "pointer",
          "flex": 1,
          "justifyContent": "center",
          "paddingRight": "0",
        }
      }
      type="button"
    >
      <p>
        4
      </p>
    </button>
    <button
      className="button"
      data-testid="button"
      onClick={[Function]}
      onKeyDown={[Function]}
      style={
        Object {
          "backgroundColor": "#e0e0e0",
          "color": "#000",
          "cursor": "pointer",
          "flex": 1,
          "justifyContent": "center",
          "paddingRight": "0",
        }
      }
      type="button"
    >
      <p>
        5
      </p>
    </button>
    <button
      className="button"
      data-testid="button"
      onClick={[Function]}
      onKeyDown={[Function]}
      style={
        Object {
          "backgroundColor": "#e0e0e0",
          "color": "#000",
          "cursor": "pointer",
          "flex": 1,
          "justifyContent": "center",
          "paddingRight": "0",
        }
      }
      type="button"
    >
      <p>
        6
      </p>
    </button>
    <button
      className="button"
      data-testid="button"
      onClick={[Function]}
      onKeyDown={[Function]}
      style={
        Object {
          "backgroundColor": "#f5913e",
          "color": "#000",
          "cursor": "pointer",
          "flex": 1,
          "justifyContent": "center",
          "paddingRight": "0",
        }
      }
      type="button"
    >
      <p>
        -
      </p>
    </button>
  </div>
  <div
    className="row"
  >
    <button
      className="button"
      data-testid="button"
      onClick={[Function]}
      onKeyDown={[Function]}
      style={
        Object {
          "backgroundColor": "#e0e0e0",
          "color": "#000",
          "cursor": "pointer",
          "flex": 1,
          "justifyContent": "center",
          "paddingRight": "0",
        }
      }
      type="button"
    >
      <p>
        1
      </p>
    </button>
    <button
      className="button"
      data-testid="button"
      onClick={[Function]}
      onKeyDown={[Function]}
      style={
        Object {
          "backgroundColor": "#e0e0e0",
          "color": "#000",
          "cursor": "pointer",
          "flex": 1,
          "justifyContent": "center",
          "paddingRight": "0",
        }
      }
      type="button"
    >
      <p>
        2
      </p>
    </button>
    <button
      className="button"
      data-testid="button"
      onClick={[Function]}
      onKeyDown={[Function]}
      style={
        Object {
          "backgroundColor": "#e0e0e0",
          "color": "#000",
          "cursor": "pointer",
          "flex": 1,
          "justifyContent": "center",
          "paddingRight": "0",
        }
      }
      type="button"
    >
      <p>
        3
      </p>
    </button>
    <button
      className="button"
      data-testid="button"
      onClick={[Function]}
      onKeyDown={[Function]}
      style={
        Object {
          "backgroundColor": "#f5913e",
          "color": "#000",
          "cursor": "pointer",
          "flex": 1,
          "justifyContent": "center",
          "paddingRight": "0",
        }
      }
      type="button"
    >
      <p>
        +
      </p>
    </button>
  </div>
  <div
    className="row"
  >
    <button
      className="button"
      data-testid="button"
      onClick={[Function]}
      onKeyDown={[Function]}
      style={
        Object {
          "backgroundColor": "#e0e0e0",
          "color": "#000",
          "cursor": "pointer",
          "flex": 2,
          "justifyContent": "center",
          "paddingRight": "0",
        }
      }
      type="button"
    >
      <p>
        0
      </p>
    </button>
    <button
      className="button"
      data-testid="button"
      onClick={[Function]}
      onKeyDown={[Function]}
      style={
        Object {
          "backgroundColor": "#e0e0e0",
          "color": "#000",
          "cursor": "pointer",
          "flex": 1,
          "justifyContent": "center",
          "paddingRight": "0",
        }
      }
      type="button"
    >
      <p>
        .
      </p>
    </button>
    <button
      className="button"
      data-testid="button"
      onClick={[Function]}
      onKeyDown={[Function]}
      style={
        Object {
          "backgroundColor": "#f5913e",
          "color": "#000",
          "cursor": "pointer",
          "flex": 1,
          "justifyContent": "center",
          "paddingRight": "0",
        }
      }
      type="button"
    >
      <p>
        =
      </p>
    </button>
  </div>
</div>
`;
